### PR TITLE
feat(#4075): Improve the validation error messages

### LINKED ISSUE #4075: Improve the validation error messages
As mentioned in issue https://github.com/javaparser/javaparser/issues/4064 we can improve the error messages delivered by validators to provide clear guidance on configuring the JavaParser library to use Java language level and would assist developers facing these issues.

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Improve the validation error messages

Fixes #4075.

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@jlerbsc Could you have a look, please?

File: javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java [hunk 1]
```diff
@@ -138,4 +139,19 @@ public Java1_0Validator() {
...
+
+    /**
+     * Compose a validator message with suggestion to enable a feature.
+     * @param cause Original message - cause of the error.
+     * @param level Language level to enable.
+     * @return Composed message.
+     */
+    private static String message(final String cause, final ParserConfiguration.LanguageLevel level){
+        return String.format(
+            "%s To enable this feature, add this line before parsing the Java file: StaticJavaParser.getParserConfiguration().setLanguageLevel(%s).",
```
Comment (jlerbsc) [inline javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java]: Hi, Thank you for your rpoposal but StaticJavaParser is a utility class that cannot be pushed forward. It should simply be specified that the language level must be configured in the configuration before parsing the source files.

File: javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java [hunk 2]
```diff
@@ -138,4 +139,19 @@ public Java1_0Validator() {
...
         add(noPermitsListInClasses);
     }
+
+    /**
+     * Compose a validator message with suggestion to enable a feature.
+     * @param cause Original message - cause of the error.
+     * @param level Language level to enable.
+     * @return Composed message.
+     */
+    private static String message(final String cause, final ParserConfiguration.LanguageLevel level){
```
Comment (jlerbsc) [inline javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java]: Why a private static method?

**Comment by jlerbsc** [comment]
For completeness there are also other validators whose messages could be improved.

File: javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java (hunk 1 above)
Comment (volodya-lombrozo) [reply javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java]: @jlerbsc I changed the message description. Thank you.

File: javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java (hunk 2 above)
Comment (volodya-lombrozo) [reply javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/Java1_0Validator.java]: @jlerbsc Moved into a separate class `UpgradeJavaMessage` in order to use it in other Validators.

**Comment by volodya-lombrozo** [comment]
@jlerbsc Could you take a look one more time, please? I tried to check all the validators and it seems that I applied the message for all possible places. Please, correct me if I'm wrong and I will add it.

**Comment by jlerbsc** [comment]
Before pushing the PR you should have run a build of the project to verify that the tests were still working.

**Comment by volodya-lombrozo** [comment]
@jlerbsc I've repaired the tests. Could you take a look one more time, please?

**Comment by jlerbsc** [comment]
Have you run the tests? There are still errors in the unit tests!

**Comment by volodya-lombrozo** [comment]
@jlerbsc My fault, sorry, missed several tests. Now I've checked them (by using `mvn clean install` command):
```sh
[INFO] Reactor Summary for javaparser-parent 3.25.6-SNAPSHOT:
[INFO] 
[INFO] javaparser-parent .................................. SUCCESS [  0.215 s]
[INFO] javaparser-core .................................... SUCCESS [ 12.536 s]
[INFO] javaparser-core-testing ............................ SUCCESS [ 59.468 s]
[INFO] javaparser-core-testing-bdd ........................ SUCCESS [ 10.444 s]
[INFO] javaparser-core-generators ......................... SUCCESS [  1.268 s]
[INFO] javaparser-core-metamodel-generator ................ SUCCESS [  0.163 s]
[INFO] javaparser-core-serialization ...................... SUCCESS [  1.175 s]
[INFO] javaparser-symbol-solver-core ...................... SUCCESS [  2.043 s]
[INFO] javaparser-symbol-solver-testing ................... SUCCESS [01:17 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:44 min
[INFO] Finished at: 2023-09-09T08:44:59+03:00
[INFO] ------------------------------------------------------------------------
```
Seems, now it is ok.

**Comment by jlerbsc** [comment]
Thank you for this contribution.

**Pull request merged**